src/lib.rs

```diff
@@ -1,23 +1,87 @@
-use std::error::Error;
 use clap::Parser;
+use futures::try_join;
+use reqwest::{Client, Response};
+use serde_json::{json, Value};
+use std::error::Error;
+use std::process;
 
 #[derive(Parser, Debug)]
 pub struct Args {
-    #[arg(short,long, help("Message to encrypt"))]
-    pub encrypt: Option<String>, 
-    #[arg(short,long, help("Link to decrypt message from"))]
+    #[arg(short, long, help("Message to encrypt"))]
+    pub encrypt: Option<String>,
+
+    #[arg(
+        short,
+        long,
+        help("Time of expiry after encrypting a message"),
+        default_value_t = 50
+    )]
+    pub time_to_live: u8,
+
+    #[arg(short, long, help("Room id to decrypt message from"))]
     pub decrypt: Option<String>,
 }
 
-pub fn run(args: Args) -> Result<(), Box<dyn Error>> {
-   if let Some(message) = args.encrypt {
-       println!("Encrypting {message}");
-   }
+async fn handle_response(response: Response) -> Result<String, Box<dyn Error>> {
+    if response.status().is_success() {
+        Ok(response.text().await?)
+    } else {
+        Ok(format!(
+            "Request failed with exit code: {}",
+            response.status()
+        ))
+    }
+}
+
+async fn get_secret(client: &Client, room_id: String) -> Result<String, Box<dyn Error>> {
+    let response = (*client)
+        .get(format!("http://0.0.0.0:4000/api/secrets/{}", room_id))
+        .send()
+        .await?;
+    handle_response(response).await
+}
+
+async fn submit_secret(
```

Comment:
We can move these two functions to their own module, maybe `api.rs`, and have them be defined in a `impl` block for a `APIClient` struct. You can think of an `impl` block like a Java class where you defined the functions that implementation contains

We can have a struct that wraps the `reqwest` Client
```rust
pub struct APIClient {
    client: Client,
}
```

And define some functions for that struct, like
```rust
impl APIClient {
    pub fn new() -> Self {
        Self { client: Client::new() }
    }

    pub async fn get_secret(&self, room_id: String) -> Result<String, Box<dyn Error>> {
         ....
    }

    pub async fn get_secret(&self, room_id: String) -> Result<String, Box<dyn Error>> {
         ....
    }
}
```

Note that I replaced `client: &Client` with `&self`. It's simply shorthand notation, we still need to pass a client when calling the functions